### ISSUE TITLE
Qt 6 dependency is detected, but Qt 5 gets deployed

### ISSUE THREAD

**Issue by VioletGiraffe** [probonopd/go-appimage#268]
The problem is well illustrated by this log. Qt 6 dependency is detected (correctly), but Qt 5 is deployed. Looks like it could be because Qt 5 was the first folder where it found ` libqxcb.so`?..
```
2023/11/16 23:34:41 Detected Qt 6
2023/11/16 23:34:41 Offset of qt_prfxpath: 3522924
2023/11/16 23:34:41 Length of value of qt_prfxpath: 4
2023/11/16 23:34:41 qt_prfxpath: /usr
2023/11/16 23:34:41 Got qt_prfxpath but it does not contain 'plugins'
2023/11/16 23:35:28 libqxcb.so found: [/usr/lib/aarch64-linux-gnu/qt5/plugins/platforms/libqxcb.so /usr/lib/aarch64-linux-gnu/qt6/plugins/platforms/libqxcb.so]
2023/11/16 23:35:28 Guessed qt_prfxpath to be /usr/lib/aarch64-linux-gnu/qt5
2023/11/16 23:35:28 Looking in /usr/lib/aarch64-linux-gnu/qt5/plugins
```

**Comment by probonopd** [comment]
> Guessed qt_prfxpath to be /usr/lib/aarch64-linux-gnu/qt5

This is the issue. It is wrong (assumig you want Qt6). We need to fx it.

**Comment by VioletGiraffe** [comment]
That is a good question, I navigated the Qt folders a bit and it all looks like a bunch of mess. But it does work. How do the Qt .so libraries know where to look for plugins? They're not in the same parent folders as it is the case on Windows.

**Comment by probonopd** [comment]
The string `qt_prfxpath` is hardcoded in one of the Qt .so libraries. It's pretty much under-documented, though.

**Comment by VioletGiraffe** [comment]
Okay, judging from the `appimagetool's log the string says just `/usr`, that's wrong as there are no plugins there. But somehow Qt works fine.

**Comment by probonopd** [comment]
Black magic!

**Comment by probonopd** [comment]
> So the question is, as far as I understand, how to locate the plugins. They're not in /usr/lib/aarch64-linux-gnu/ but instead in /usr/lib/aarch64-linux-gnu/qt6/.

... but currently go-appimage (wrongly)

> Guessed qt_prfxpath to be /usr/lib/aarch64-linux-gnu/qt5

**Comment by probonopd** [comment]
All I can tell you at the moment is that the bug is somewhere in 

https://github.com/probonopd/go-appimage/blob/53294ec0fa99b89babd598e087e859259ed32625/src/appimagetool/appdirtool.go#L1471-L1490

**Comment by probonopd** [comment]
In the meantime, try `export QTDIR=/usr/lib/aarch64-linux-gnu/qt6/` before running `linuxdeployqt`. Do this on a new AppDir (one which `linuxdeployqt` was never run on before).

**Comment by VioletGiraffe** [comment]
Why new? Do you store something that affects consequent runs? Does nuking and re-creating the folder count as "new"?

**Comment by VioletGiraffe** [comment]
> In the meantime, try `export QTDIR=/usr/lib/aarch64-linux-gnu/qt6/` before running `linuxdeployqt`.

That worked! The correct paths are in the log now, and the AppImage size is reduced. However, it still doesn't run:
```
Failed to load the image at
Press any key to exit.
```
Any ideas? What can I do to troubleshoot this?

**Comment by probonopd** [comment]
Is that message coming from your program?

**Comment by VioletGiraffe** [comment]
No, I'm pretty sure it's coming from AppRun. This is a typical message for a malformed AppImage.

**Comment by VioletGiraffe** [comment]
You're right, sincere apologies. I was so puzzled by the message that I was certain it's from the AppImage runtime.
And many thanks for your help!
